nomenclature/codelist.py

```diff
@@ -364,6 +335,35 @@ def read_excel(cls, name, source, sheet_name, col, attrs=None):
 
         return cls(name=name, mapping=mapp)
 
+    def check_illegal_characters(
+        self, config: NomenclatureConfig = None
+    ) -> Dict[str, Code]:
+        """Check that no illegal characters are left in codes after tag replacement"""
+        forbidden = ["{", "}"]
```

Comment:
As mentioned below as well, I would just extend the list of forbidden characters directly. Unless, of course there is a need for project specific banned characters.